vmdb/app/models/ems_kubernetes.rb

```diff
@@ -20,7 +27,8 @@ def self.raw_connect(hostname, port, api_version)
   def self.raw_api_endpoint(hostname, port)
     require 'uri'
 
-    uri = URI::HTTP.build(:path => "/api", :port => port.to_i)
+    port &&= port.to_i
+    uri = URI::HTTP.build(:path => "/api", :port => port)
```

Comment:
Cool. Looks like this still works with rails 4.0's try.

`NilClass#try` is defined as `def try ; nil ; end`
`Object#try` has changed in rails 4.0 (as brandon has pointed out) to now have something like a `responds_to?` in it.